docs/gen-ai/gen-ai-events.md

```diff
@@ -0,0 +1,379 @@
+
+### `ToolCall` object
+
+| Body Field | Type                           | Description                            | Examples                                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|------------|--------------------------------|----------------------------------------|-------------------------------------------------|-------------------|
+| `id`       | string                         | The id of the tool call                | `call_mszuSIzqtI65i1wAUOE8w5H4`                 | `Required`        |
+| `type`     | string                         | The type of the tool                   | `function`                                      | `Required`        |
+
+### `Function` object
+
+| Body Field  | Type       | Description                            | Examples                   | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|-------------|------------|----------------------------------------|----------------------------|-------------------|
+| `name`      | string     | The name of the function to call       | `get_link_to_otel_semconv` | `Required`        |
+| `arguments` | `AnyValue` | The arguments to pass the the function | `{"semconv": "gen_ai"}`    | `Opt-In`          |
+
+## Tool message
+
+This event describes the output of the tool or function submitted back to the model.
+
+The event name MUST be `gen_ai.tool.message`.
+
+| Body Field     | Type   | Description                                   | Examples                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|----------------|--------|-----------------------------------------------|---------------------------------|-------------------|
+| `content`      | string | The contents of the tool message.             | `opentelemetry.io`              | `Opt-In`          |
+| `tool_call_id` | string | Tool call that this message is responding to. | `call_mszuSIzqtI65i1wAUOE8w5H4` | `Required`        |
+
+## Choice message
+
+This event describes the model-generated chat response message (choice).
+If GenAI model returns multiple choices, each choice SHOULD be recorded as an individual event.
+
+When response is streamed, instrumentations that report response events MUST reconstruct and report the full message and MUST NOT report individual chunks as events.
+If the request to GenAI model fails with an error before content is received, instrumentation SHOULD report an event with truncated content (if enabled). If `finish_reason` was not received, it MUST be set to `error`.
+
+The event name MUST be `gen_ai.choice`.
```

Comment:
```suggestion
The event name MUST be `gen_ai.choice.message`.
```

Reply:
I wanted to have a different name pattern for choice, because choice is not a message - it wraps assistant message and includes some metadata about it.

E.g.

```json
{
    "index": 0,
    "finish_reason": "stop"
    "message": {
      "role": "assistant",
      "content": "\n\nHello there, how may I assist you today?",
    },
  }
```

I feel calling it a `message` would be misleading. 

Happy to hear any suggestions or thoughts

Reply:
I don't have a strong opinion either way. It felt like an accidental omission given the "Naming pattern" section above:

> The generic events applicable to multiple GenAI models follow `gen_ai.{type}.message` naming pattern.
> GenAI vendor-specific instrumentations SHOULD follow `gen_ai.{gen_ai.system}.{type}.message` pattern to record system-specific events.

and this section is called "Choice message".

I agree with you that having a "message" Body property of an event named `(something).message` would be odd.

Reply:
good catch on the naming pattern - I'll try to rephrase it

Reply:
I renamed "naming pattern" to "custom events" and removed general part - there is no need to describe naming pattern for existing events. I kept the suggestion for vendor-specific ones. LMK if it helps